### ISSUE TITLE
Update from 6.6.7 to 6.7.1 break REST API

### ISSUE THREAD

**Issue by michel-hete** [pimcore/pimcore#7091]
```
protected function getDataDocument($id)
    {
        $document = $this->loadDocument($id);

        $this->checkElementPermission($document, 'get');

        $type = $document->getType();
        $getter = sprintf('getDocument%sById', ucfirst($type));
        $object = null;

        if (method_exists($this->service, $getter)) {
            $object = $this->service->$getter($id);
        } else {
            // check if the getter is implemented by a plugin
            $class = '\\Pimcore\\Model\\Webservice\\Data\\Document\\' . ucfirst($type) . '\\Out';

            if (Tool::classExists($class)) {
                Document\Service::loadAllDocumentFields($document);
                $object = Webservice\Data\Mapper::map($document, $class, 'out');
            } else {
                return null;
            }
        }

        return $object;
    }

```
In 6.6.7, this method is OK 

**Pimcore updated to 6.7.1, the same call on the same method is KO:**

{
    "success": false,
    "msg": "Too few arguments to function Pimcore\\Model\\Element\\ElementDescriptor::__construct(), 0 passed in /data/pimcore/vendor/pimcore/pimcore/models/Webservice/Data/Mapper.php on line 94 and exactly 2 expected",
    "trace": [
        {
            "file": "/data/pimcore/vendor/pimcore/pimcore/models/Webservice/Data/Mapper.php",
            "line": 94,
            "function": "__construct",
            "class": "Pimcore\\Model\\Element\\ElementDescriptor",
            "type": "->",
            "args": []
        },
        {
            "file": "/data/pimcore/vendor/pimcore/pimcore/models/Document/Tag.php",
            "line": 642,
            "function": "map",
            "class": "Pimcore\\Model\\Webservice\\Data\\Mapper",
            "type": "::",
            "args": [
                {},
                "Pimcore\\Model\\Element\\ElementDescriptor",
                "out"
            ]
        },
        {
            "file": "/data/pimcore/vendor/pimcore/pimcore/models/Webservice/Data/Document/PageSnippet.php",
            "line": 66,
            "function": "getForWebserviceExport",
            "class": "Pimcore\\Model\\Document\\Tag",
            "type": "->",
            "args": []
        },
        {
            "file": "/data/pimcore/vendor/pimcore/pimcore/models/Webservice/Data/Mapper.php",
            "line": 96,
            "function": "map",
            "class": "Pimcore\\Model\\Webservice\\Data\\Document\\PageSnippet",
            "type": "->",
            "args": [
                {
                    "scheduledTasks": []
                },
                null
            ]
        },
        {
            "file": "/data/pimcore/vendor/pimcore/pimcore/models/Webservice/Service.php",
            "line": 158,
            "function": "map",
            "class": "Pimcore\\Model\\Webservice\\Data\\Mapper",
            "type": "::",
            "args": [
                {
                    "scheduledTasks": []
                },
                "\\Pimcore\\Model\\Webservice\\Data\\Document\\Page\\Out",
                "out"
            ]
        },

**Comment by brusch** [comment]
@michel-hete can you please be a bit more specific and describe how to reproduce the issue? Thanks!

**Comment by michel-hete** [comment]
Sorry, i can't because i wrote some new rest controller (class CustomDocumentRestController extends AbstractElementController for example).  I don't know to reproduce that with a stock pimcore.

The main problem is in the new "ElementDescriptor::__construct()". 
He need 2 parameters. and the "Mapper.php" create instance without parameters.

Retry to test your rest API...

I hope you will found a correction for this problem.

**Comment by weisswurstkanone** [comment]
Unable to reproduce with sample data from demo instance and then calling
/webservice/rest/document/id/38

Maybe related to your custom implementation? Please try to isolate the problem. Maybe it is related to a certain data type?

**Comment by weisswurstkanone** [comment]
@michel-hete ping

**Comment by weisswurstkanone** [comment]
@michel-hete any new findings?

**Comment by michel-hete** [comment]
Hello.

No.
No need to use 6.7 version, i wait on 6.6.7.
No time to look this problem and to proceed to more tests.
Sorry.

Michel .
De : Josef Aichhorn <notifications@github.com>
Envoyé : jeudi 5 novembre 2020 11:41
À : pimcore/pimcore <pimcore@noreply.github.com>
Cc : Michel Hete <mhete@jouve.com>; Mention <mention@noreply.github.com>
Objet : Re: [pimcore/pimcore] Update from 6.6.7 to 6.7.1 break REST API (#7091)


@michel-hete<https://fra01.safelinks.protection.outlook.com/?url=https%3A%2F%2Fgithub.com%2Fmichel-hete&data=04%7C01%7Cmhete%40jouve.com%7Ce44ee9a674374989a12908d881773c50%7C10e992ff068b4af381f13cc921376064%7C1%7C0%7C637401696393802353%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C1000&sdata=4cPq3buIQwFxBAuToLPazs1g7sE5mG%2FcJEFMaXNp8aI%3D&reserved=0> any new findings?

**Comment by weisswurstkanone** [comment]
Thanks for the feedback! I'll close this for now. Please feel free to open a new ticket in case you face this issue again.
